src/components/Search/SearchResultsListItem.tsx

```diff
@@ -12,22 +12,16 @@ import { truncateHash, getLatestCommitMessage } from '../../utils/helpers';
 function SearchResultsListItem(props: SearchResultsListItemProps) {
   const { index, item } = props;
   const isChecked: boolean = useAppSelector((state) =>
-    state.checkedRevisions.revisions.includes(index),
+    state.checkedRevisions.revisions.includes(item),
   );
   const { handleToggle } = useCheckRevision();
 
-  const indexString = index.toString();
-
   const revisionHash = truncateHash(item.revision);
   const commitMessage = getLatestCommitMessage(item);
 
   return (
     <>
-      <ListItemButton
-        key={item.id}
-        id={indexString}
```

Comment:
`id` was used by the `handleToggle` hook to check/uncheck a result. Now we are passing the revision itself to store in checkedRevisions.